tests/model/test_model.py

```diff
@@ -888,24 +907,54 @@ def test_update_private_message(
                     "topic": "party",
                 },
                 "lets_party",
+                "party",
+                "stream",
+                "change_all",
                 True,
             ),
         ],
     )
     def test_update_stream_message(
-        self, mocker, model, response, return_value, req, old_topic, footer_updated
+        self,
+        mocker,
+        model,
+        response,
+        return_value,
+        req,
+        old_topic,
+        new_topic,
+        stream_name,
+        propagate_mode,
+        footer_updated,
     ):
         self.client.update_message = mocker.Mock(return_value=response)
         model.index["messages"][req["message_id"]]["subject"] = old_topic
+        req["topic"] = new_topic
+        model.index["messages"][req["message_id"]]["display_recipient"] = stream_name
+        req["propagate_mode"] = propagate_mode
+
+        recent_moved_msgs = ""
+        if propagate_mode == "change_all":
+            recent_moved_msgs = "all"
+        if propagate_mode == "change_later":
+            recent_moved_msgs = "some"
+        if propagate_mode == "change_one":
+            recent_moved_msgs = "one"
```

Comment:
It's better to put the explicit text into the test cases above, than have conditional code here - particularly if it is exactly the same as the code in the application.